src/web.rs

```diff
@@ -91,6 +144,37 @@ impl WebImpl {
             ImageData::new_with_u8_clamped_array(Clamped(&bitmap), width.into()).unwrap();
 
         // This can only throw an error if `data` is detached, which is impossible.
-        self.ctx.put_image_data(&image_data, 0.0, 0.0).unwrap();
+        match self {
+            WebImpl::Canvas { ctx, .. } => ctx.put_image_data(&image_data, 0.0, 0.0),
+            WebImpl::OffscreenCanvas { ctx, .. } => ctx.put_image_data(&image_data, 0.0, 0.0),
+        }
+        .unwrap()
+    }
+}
+
+/// Extension methods for the Wasm target on [`Surface`](crate::Surface).
+pub trait SurfaceExtWeb: Sized {
+    /// Creates a new instance of this struct, using the provided [`HtmlCanvasElement`].
+    fn from_canvas(canvas: HtmlCanvasElement) -> Result<Self, SoftBufferError>;
+
+    /// Creates a new instance of this struct, using the provided [`HtmlCanvasElement`].
+    fn from_offscreen_canvas(offscreen_canvas: OffscreenCanvas) -> Result<Self, SoftBufferError>;
```

Comment:
I'd still like to avoid having `web_sys` as a public dependency if possible. I know `winit` does it, but I'd like to keep the number of public dependencies in this project as low as possible.

Would it be possible to just pass in the ID/XPath of the canvas into these functions instead?

Reply:
That sounds reasonable to me